### PR TITLE
feat(1271): extend query options

### PR DESCRIPTION
**Pull request by DekusDenial**
extending query options to datastore
related to: https://github.com/screwdriver-cd/screwdriver/issues/1271

### CODE REVIEW

File: models/template.js [hunk 1]
```diff
@@ -34,8 +24,15 @@ const MODEL = {
     version: Template.version,
     description: Template.description,
     maintainer: Template.maintainer,
+    pipelineId,
+    namespace: Template.namespace,
     images: Template.images,
-    pipelineId
+    createTime: Joi
```
Comment (tkyi) [inline models/template.js]: any reason why the order of these fields were changed?

File: plugins/datastore.js [hunk 2]
```diff
@@ -33,7 +33,9 @@ const SCHEMA_SCAN = Joi.object().keys({
         keyword: Joi.string().max(200).required()
     }),
     sort: Joi.string().lowercase().valid(['ascending', 'descending']).default('descending'),
-    sortBy: Joi.string().max(100)
+    sortBy: Joi.string().max(100),
+    exclude: Joi.array().items(Joi.string()),
```
Comment (tkyi) [inline plugins/datastore.js]: might be better to put max as well, as in `Joi.array().items(Joi.string().max(100)),`

File: models/command.js [hunk 3]
```diff
@@ -10,25 +10,23 @@ const MODEL = {
...
     version: Command.version,
     description: Command.description,
-    usage: Command.usage,
     maintainer: Command.maintainer,
     format: Command.format,
     habitat: Command.habitat,
     docker: Command.docker,
     binary: Command.binary,
-    pipelineId
+    name: Command.name,
```
Comment (tkyi) [inline models/command.js]: any reason to change the order of fields?

Comment (DekusDenial) [reply models/command.js]: same as above

**Pull request merged**